cli/src/cluster_query.rs

```diff
@@ -1791,53 +1823,74 @@ pub fn process_show_stakes(
         ..RpcProgramAccountsConfig::default()
     };
 
-    if let Some(vote_account_pubkeys) = vote_account_pubkeys {
-        // Use server-side filtering if only one vote account is provided
-        if vote_account_pubkeys.len() == 1 {
+    // withdrawer filter
+    let withdraw_authority_filter = withdraw_authority_pubkey
+    .map(|pubkey|  
+        RpcFilterType::Memcmp(Memcmp::new_base58_encoded(
+            44, 
+            pubkey.as_ref()
+        ))
+    );
+
+    let stake_account_progress_bar = new_spinner_progress_bar();
+    stake_account_progress_bar.set_message("Fetching stake accounts...");
+
+    let mut all_stake_accounts = Vec::new();
+    if vote_account_pubkeys.is_empty() {
+        if let Some(withdraw_authority_filter) = &withdraw_authority_filter {
+            program_accounts_config.filters.as_mut().unwrap().push(
+                withdraw_authority_filter.clone()
+            );
+        }
+        all_stake_accounts.append(
+            &mut rpc_client.get_program_accounts_with_config(
+                &stake::program::id(), 
+                program_accounts_config
+            )?
+        );
+    } else {
+        for vote_account_pubkey in &vote_account_pubkeys {
             program_accounts_config.filters = Some(vec![
                 // Filter by `StakeStateV2::Stake(_, _)`
                 RpcFilterType::Memcmp(Memcmp::new_base58_encoded(0, &[2, 0, 0, 0])),
                 // Filter by `Delegation::voter_pubkey`, which begins at byte offset 124
                 RpcFilterType::Memcmp(Memcmp::new_base58_encoded(
                     124,
-                    vote_account_pubkeys[0].as_ref(),
+                    Pubkey::from_str(vote_account_pubkey).unwrap().as_ref(),
                 )),
             ]);
+            if let Some(withdraw_authority_filter) = &withdraw_authority_filter {
+                program_accounts_config.filters.as_mut().unwrap().push(
+                    withdraw_authority_filter.clone()
+                );
+            }
+            all_stake_accounts.append(
+                &mut rpc_client.get_program_accounts_with_config(
+                    &stake::program::id(), 
+                    program_accounts_config.clone()
+                )?
```

Comment:
Unfortunately, `get_program_accounts` calls are very expensive; it is not feasible to call this multiple times with different server-side filters. This needs to be called only once and filtered client-side.

Reply:
got it, but how can I get all stake accounts with their delegate specified? As far as I can see, `rpc_client.get_program_accounts_with_config()` contains only the owner of the stake account.

Reply:
Fixed, now uses only one call. Also, made sure to use `fmt `and `clippy`